### PR TITLE
Make actions and constants strongly typed. No functional changes

### LINKED ISSUE #1153: convert stuff in constants.ts to enums
including currentScreen to an int or whatever

### PR DESCRIPTION
**Pull request by robertos**
Fixes #1153 

Also changed some interfaces to straight up types. Didn't do this replacement everywhere though.
Main benefit: now actions are strongly typed, so it won't compile with typos on action names.

### CODE REVIEW

File: public/js/src/game/ui/action.ts [hunk 1]
```diff
@@ -58,7 +60,7 @@ export default (data: any) => {
 };
 
 // Define a command handler map
-const actionFunctions = new Map();
+const actionFunctions = new Map<ActionIncludingHypothetical['type'], any>();
```
Comment (robertos) [inline public/js/src/game/ui/action.ts:63]: #magic

File: public/js/src/constants.ts [hunk 2]
```diff
@@ -27,39 +27,40 @@ export const ARROW_COLOR = {
...
-  EFFICIENCY: -4,
-  MIN_EFFICIENCY: -5,
-};
-export const STACK_DIRECTION = { // Used in the "Up or Down" and "Reversed" variants
-  UNDECIDED: 0,
-  UP: 1,
-  DOWN: 2,
-  FINISHED: 3,
-};
+export enum ActionType {
```
Comment (Zamiell) [inline public/js/src/constants.ts:30]: airbnb guide says that constants should be uppercase with underscores

Comment (robertos) [reply public/js/src/constants.ts:30]: This is not a constant anymore. It's a type now

Comment (robertos) [reply public/js/src/constants.ts:30]: https://basarat.gitbook.io/typescript/styleguide#enum

**Pull request merged**